src/test/java/com/orbitz/consul/CatalogTests.java

```diff
@@ -182,4 +183,86 @@ public void shouldDeregisterWithDefaultDC() throws InterruptedException {
 
         assertFalse(found);
     }
+
+    @Test
+    public void shouldGetServicesInCallback() throws ExecutionException, InterruptedException, TimeoutException {
+        CatalogClient catalogClient = client.catalogClient();
+
+        String serviceName = UUID.randomUUID().toString();
+        String serviceId = createAutoDeregisterServiceId();
+        client.agentClient().register(20001, 20, serviceName, serviceId);
+
+        CompletableFuture<Map<String, List<String>>> cf = new CompletableFuture<>();
+        catalogClient.getServices(QueryOptions.BLANK, callbackFuture(cf));
+
+        Map<String, List<String>> result = cf.get(1, TimeUnit.SECONDS);
+
+        assertTrue(result.containsKey(serviceName));
+    }
+
+    @Test
+    public void shouldGetServiceInCallback() throws ExecutionException, InterruptedException, TimeoutException {
+        CatalogClient catalogClient = client.catalogClient();
+
+        String serviceName = UUID.randomUUID().toString();
+        String serviceId = createAutoDeregisterServiceId();
+        client.agentClient().register(20001, 20, serviceName, serviceId);
+
+        CompletableFuture<List<CatalogService>> cf = new CompletableFuture<>();
+        catalogClient.getService(serviceName, QueryOptions.BLANK, callbackFuture(cf));
+
+        List<CatalogService> result = cf.get(1, TimeUnit.SECONDS);
+
+        assertEquals(1, result.size());
+        CatalogService service = result.get(0);
+
+        assertEquals(serviceId, service.getServiceId());
+    }
+
+    @Test
+    public void shouldGetNodeInCallback() throws ExecutionException, InterruptedException, TimeoutException {
```

Comment:
this is not related to the review (but quite useful), please push it in a later review

Reply:
in this PR added method with callback in CatalogClient